src/account/mod.rs

```diff
@@ -242,6 +250,127 @@ pub struct Account {
     has_pending_changes: bool,
 }
 
+/// A thread guard over an account.
+#[derive(Debug, Clone)]
+pub struct AccountGuard(Arc<RwLock<Account>>);
+
+impl From<Account> for AccountGuard {
+    fn from(account: Account) -> Self {
+        Self(Arc::new(RwLock::new(account)))
+    }
+}
+
+impl Deref for AccountGuard {
+    type Target = RwLock<Account>;
+    fn deref(&self) -> &Self::Target {
+        &self.0.deref()
+    }
+}
+
+macro_rules! guard_field_getters {
+    ($ty:ident, $(#[$attr:meta] => $x:ident => $ret:ty),*) => {
+        impl $ty {
+            $(
+                #[$attr]
+                pub fn $x(&self) -> $ret {
+                    let account = self.read().unwrap();
+                    account.$x().clone()
+                }
+            )*
+        }
+    }
+}
+
+guard_field_getters!(
+    AccountGuard,
+    #[doc = "Bridge to [Account#id](struct.Account.html#method.id)."] => id => AccountIdentifier,
+    #[doc = "Bridge to [Account#signer_type](struct.Account.html#method.signer_type)."] => signer_type => SignerType,
+    #[doc = "Bridge to [Account#index](struct.Account.html#method.index)."] => index => usize,
+    #[doc = "Bridge to [Account#alias](struct.Account.html#method.alias)."] => alias => String,
+    #[doc = "Bridge to [Account#created_at](struct.Account.html#method.created_at)."] => created_at => DateTime<Utc>,
+    #[doc = "Bridge to [Account#messages](struct.Account.html#method.messages).
+    This method clones the addresses so prefer the using the `read` method to access the account instance."] => messages => Vec<Message>,
+    #[doc = "Bridge to [Account#addresses](struct.Account.html#method.addresses).
+    This method clones the addresses so prefer the using the `read` method to access the account instance."] => addresses => Vec<Address>,
+    #[doc = "Bridge to [Account#client_options](struct.Account.html#method.client_options)."] => client_options => ClientOptions
+);
+
+impl AccountGuard {
+    /// Returns the builder to setup the process to synchronize this account with the Tangle.
+    pub fn sync(&self) -> AccountSynchronizer {
+        AccountSynchronizer::new(self.clone())
+    }
+
+    /// Gets a new unused address and links it to this account.
+    pub fn generate_address(&self) -> crate::Result<Address> {
+        let mut account = self.0.write().unwrap();
+        let address = crate::address::get_new_address(&account)?;
+        account.addresses.push(address.clone());
+
+        account.has_pending_changes = true;
+
+        let _ = crate::monitor::monitor_address_balance(self.clone(), address.address());
+
+        Ok(address)
+    }
+
+    /// Bridge to [Account#latest_address](struct.Account.html#method.latest_address).
+    pub fn latest_address(&self) -> Option<Address> {
+        let account = self.0.read().unwrap();
```

Comment:
I don't think there should be any problem with perform these operations inline (i.e: without the intermediate `account` binding).

Reply:
yeah thought about it when you mentioned on the other comment :) thanks for the tip